lte/gateway/fabfile.py

```diff
@@ -59,147 +65,154 @@
+            c_gw.run('make clean')
+            build_type = "Debug" if debug_mode else "RelWithDebInfo"
+
+            build_cmd = f'./release/build-magma.sh --hash {hash}' \
+                        f' --commit-count {commit_count} --type {build_type}' \
+                        f' --cert {cert_file} --proxy {proxy_config} --os {os}'
+            # set '/usr/bin/' in PATH to ensure that the correct version of
+            # python is used
+            c_gw.run(
+                build_cmd,
+                env={'PATH': '/usr/bin:/usr/local/go/bin:/home/vagrant/go/bin:/usr/lib/ccache:$PATH'},
             )
-            run('cp *.deb ~/magma-packages')
-            run('sudo rm -f *.deb')
 
-        if all_deps:
-            pkg.download_all_pkgs()
-            run('cp /var/cache/apt/archives/*.deb ~/magma-packages')
+            c_gw.run('rm -rf ~/magma-packages')
+            c_gw.run('mkdir -p ~/magma-packages')
+            c_gw.run('cp -f ~/magma-deps/*.deb ~/magma-packages', warn=True)
+            c_gw.run('mv *.deb ~/magma-packages')
+
+            with c_gw.cd('release'):
+                mirrored_packages_file = 'mirrored_packages'
+                if os == "ubuntu":
+                    mirrored_packages_file += '_focal'
+                if vm and vm.startswith('magma_'):
+                    mirrored_packages_file += vm[5:]
+
+                c_gw.run(
+                    f'cat {mirrored_packages_file}'
+                    + ' | xargs -I% sudo aptitude download -q2 %',
+                )
+                c_gw.run('cp *.deb ~/magma-packages')
+                c_gw.run('sudo rm -f *.deb')
 
-        # Copy out C executables into magma-packages as well
-        _copy_out_c_execs_in_magma_vm()
+            if all_deps:
+                pkg.download_all_pkgs(c_gw)
+                c_gw.run('cp /var/cache/apt/archives/*.deb ~/magma-packages')
 
+            # Copy out C executables into magma-packages as well
+            _copy_out_c_execs_in_magma_vm(c_gw)
 
-def openvswitch(destroy_vm='False', destdir='~/magma-packages/'):
-    destroy_vm = strtobool(destroy_vm)
+
+@task
+def openvswitch(c, destroy_vm=False, destdir='~/magma-packages/'):
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        vagrant_setup('magma', destroy_vm=destroy_vm)
-    run('~/magma/third_party/gtp_ovs/ovs-gtp-patches/2.15/build.sh ' + destdir)
+    c_gw = vagrant_connection(c, 'magma', destroy_vm=destroy_vm)
+    with c_gw:
+        c_gw.run('~/magma/third_party/gtp_ovs/ovs-gtp-patches/2.15/build.sh ' + destdir)
 
 
-def depclean():
+@task
+def depclean(c):
     '''Remove all generated packaged for dependencies'''
     # If a host list isn't specified, default to the magma vagrant vm
-    if not env.hosts:
-        setup_env_vagrant()
-    run('rm -rf ~/magma-deps')
+    c_gw = vagrant_connection(c, 'magma')
+    with c_gw:
+        c_gw.run('rm -rf ~/magma-deps')
```

Comment:
Nit: We could make this (and the same pattern elsewhere) slightly more pythonic with 
```
with vagrant_connection(c, 'magma') as c_gw:
   c_gw.run('rm -rf ~/magma-deps')
```
Approving anyway,  feel free to implement this change or not.